### PR TITLE
Number.prototype functions should throw TypeError if 'this' is not number.

### PR DESCRIPTION
**Pull request by dbatyai**
JerryScript-DCO-1.0-Signed-off-by: Dániel Bátyai dbatyai.u-szeged@partner.samsung.com

### CODE REVIEW

File: tests/jerry/number-prototype-to-fixed.js [hunk 1]
```diff
@@ -38,40 +38,23 @@ assert((123456789012345678901.0).toFixed(20) === "123456789012345680000.00000000
...
-try {
-    assert(obj.toFixed(-Infinity));
-    assert(false);
-} catch (e) {
-    assert(e instanceof RangeError);
-}
-
-try {
-    assert(obj.toFixed(-1));
+    (12).toFixed(-1);
```
Comment (ruben-ayrapetyan) [inline tests/jerry/number-prototype-to-fixed.js]: @dbatyai, why do we need the change?

**Comment by ruben-ayrapetyan** [comment]
Beginning of current `ecma_builtin_number_prototype_object_to_string` implementation contains code very similar to `ecma_builtin_number_prototype_object_value_of` function. Maybe, we could replace it with the call to valueOf too.

**Comment by dbatyai** [comment]
@ruben-ayrapetyan, I've updated the PR.

**Comment by ruben-ayrapetyan** [comment]
Looks good to me

**Comment by zherczeg** [comment]
LGTM

**Comment by egavrin** [comment]
`make push`

**Comment by egavrin** [comment]
Merged at e5b2e1e3891bb4c78595fa9b5e617dd859f70765

**Pull request closed without merging**